compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -161,22 +205,32 @@ fn resolve_name_in_module(
             ModuleDefId::GlobalId(_) => panic!("globals cannot be in the type namespace"),
         };
 
-        current_mod = &def_maps[&new_module_id.krate].modules[new_module_id.local_id.0];
+        check_can_reference_function(
+            def_maps,
+            importing_crate,
+            starting_mod,
+            current_mod_id,
+            visibility,
+            last_segment,
+        )?;
```

Comment:
Us just erroring out here and not returning the fully resolved path _as well_ as the error makes this probably the most breaking change that we've ever done.

We're probably going to have to have this be a warning for a number of releases to avoid bricking most code out there.